creator-node/src/monitors/MonitoringQueue.js

```diff
@@ -0,0 +1,111 @@
+const Bull = require('bull')
+const redis = require('../redis')
+const config = require('../config')
+const { logger } = require('../logging')
+const { getDatabaseSize } = require('./database')
+
+const MONITORING_REDIS_PREFIX = 'monitoring'
+const PROCESS_NAMES = Object.freeze({
+  monitor: 'monitor'
+})
+
+/**
+ * List of all monitors to run, containing:
+ *  @param {string} name A unique name (for caching in redis)
+ *  @param {number} ttl TTL in seconds for how long a cached value is good for
+ *  @param {function} func The actual work to compute a value. The return value is what is cached.
+ */
+const MONITORS = [
+  {
+    name: 'databaseSize',
+    ttl: 120,
+    func: getDatabaseSize
+  }
+  // TODO: Add more monitors
+]
+
+/**
+ * A persistent cron-style queue that periodically monitors various
+ * health metrics and caches values in redis.
+ *
+ * The queue runs every minute on cron, but individual monitors establish
+ * their own freshness/refresh rate to operate on.
+ *  1. The queue spins up and for each monitor checks to see if it needs a refresh
+ *  2. Refreshes the value and stores the update in redis
+ */
+class MonitoringQueue {
+  constructor () {
+    this.queue = new Bull(
+      'monitoring-queue',
+      {
+        redis: {
+          port: config.get('redisPort'),
+          host: config.get('redisHost')
+        },
+        defaultJobOptions: {
+          removeOnComplete: true,
+          removeOnFail: true
+        }
+      }
+    )
+
+    // Clean up anything that might be still stuck in the queue on restart
+    this.queue.empty()
+
+    this.queue.process(
+      PROCESS_NAMES.monitor,
+      /* concurrency */ 1,
+      async (job, done) => {
+        this.logStatus('Starting')
+
+        // Iterate over each monitor and set a new value if the cached
+        // value is not fresh.
+        MONITORS.forEach(async monitor => {
+          try {
+            await this.setIfNotFresh(monitor)
```

Comment:
currently, if any function call fails, since there is no try-catch, the entire monitor loop will be short circuited. we should handle and maybe only set new value if it comes back not null. or even better, explicitly try catch the `monitor.func()` and only set value if no error?

Reply:
jesus lol i literally stared at this code and did not notice the try-catch